sql/core/src/main/scala/org/apache/spark/sql/jdbc/MySQLDialect.scala

```diff
@@ -217,7 +219,7 @@ private case object MySQLDialect extends JdbcDialect with SQLConfHelper {
             // The only property we are building here is `COMMENT` because it's the only one
             // we can get from `SHOW INDEXES`.
             val properties = new util.Properties();
-            if (indexComment.nonEmpty) properties.put("COMMENT", indexComment)
```

Comment:
Seems to me if `Index_comment` is not set, MySQL defaults it to empty String?

Reply:
> Yes, you are right! I investigated it carefully.
> INDEX_COMMENT nullability is NO, the default value is empty String.

mysql> desc INFORMATION_SCHEMA.STATISTICS;
+---------------+---------------+------+-----+---------+-------+
| Field         | Type          | Null | Key | Default | Extra |
+---------------+---------------+------+-----+---------+-------+
| TABLE_CATALOG | varchar(512)  | NO   |     |         |       |
| TABLE_SCHEMA  | varchar(64)   | NO   |     |         |       |
| TABLE_NAME    | varchar(64)   | NO   |     |         |       |
| NON_UNIQUE    | bigint(1)     | NO   |     | 0       |       |
| INDEX_SCHEMA  | varchar(64)   | NO   |     |         |       |
| INDEX_NAME    | varchar(64)   | NO   |     |         |       |
| SEQ_IN_INDEX  | bigint(2)     | NO   |     | 0       |       |
| COLUMN_NAME   | varchar(64)   | NO   |     |         |       |
| COLLATION     | varchar(1)    | YES  |     | NULL    |       |
| CARDINALITY   | bigint(21)    | YES  |     | NULL    |       |
| SUB_PART      | bigint(3)     | YES  |     | NULL    |       |
| PACKED        | varchar(10)   | YES  |     | NULL    |       |
| NULLABLE      | varchar(3)    | NO   |     |         |       |
| INDEX_TYPE    | varchar(16)   | NO   |     |         |       |
| COMMENT       | varchar(16)   | YES  |     | NULL    |       |
**| INDEX_COMMENT | varchar(1024) | NO   |     |         |       |**
+---------------+---------------+------+-----+---------+-------+

Reply:
I will close it, but maybe we need add some UT for it.I will open new pr for it.
> FYI: Whether it is necessary to implement createIndex & existIndex & dropIndex & listIndex in H2Dialet ?
> I found the above "possible null pointer" when implementing the method: listIndex in H2Dialect(the default value is null instead of empty string). @huaxingao

Reply:
Updated: 
> new UT for it: https://github.com/apache/spark/pull/37060

Reply:
> maybe we need add some UT for it

Sounds good to me. I will take a look at the new PR.

> Whether it is necessary to implement createIndex & existIndex & dropIndex & listIndex in H2Dialect ?

It's OK to implement these.

Reply:
@huaxingao implement for h2 dialect: https://github.com/apache/spark/pull/37112